### DISCUSSION TITLE
Vulnerability Detection not working

### DISCUSSION THREAD

**Discussion by pavlos1982** [wazuh/wazuh#13360] [General]
I have followed the instruction for both the Wazuh Server and Agent, the vulnerability detector is enabled,syscollector enable for the agent. I have reinstalled wazuh twice with unattended installation, the same issue, anyone to help? thanks

**Comment by MiguelazoDS** [comment]
Hello @pavlos1982, 

I would like to know a little bit more about your setup. 

- Are you using a cluster environment? Take into account that the vulnerability detector must be configured in those managers that have agents directly connected. 
- What OS are your agents running on? Did you enable the providers for your agents in the vulnerability detector conf? 
- Did you check the agents are active? vulnerability detector only scans active agents.
- What version of Wazuh are you using? It is worth mentioning that the manager version should be equal to or newer than the agent version.

Regards!

**Reply by pavlos1982** [reply]
Hi @MiguelazoDS,

Thank you for getting back to me, as requested:

. I installed it into one hyper VM on a single host for everything using this code: curl -sO https://packages.wazuh.com/4.3/wazuh-install.sh && sudo bash ./wazuh-install.sh -a

. I was previously using Ubuntu Server 22.04 LTS but since the new version 4.3.0 of wazuh is not on the compatibility list, I downgraded to ubuntu 20.04.4 server 

. all the agents are active, when I click on the vulnerability in the dashboard it shows it has scanned but there is nothing

. I am using the latest version 4.3.0

I have attached a few screenshots to reflect on my feedback, thanks

![Screenshot 2022-05-09 074959 1](https://user-images.githubusercontent.com/5363713/167338236-5385605c-e40d-4520-b91c-e36c130c0637.jpg)

![Screenshot 2022-05-09 075751 2](https://user-images.githubusercontent.com/5363713/167338883-7248b732-00bb-4df2-b9c2-b29b56c037b7.jpg)

**Reply by MiguelazoDS** [reply]
Hi @pavlos1982, 

You're right about the support for Ubuntu Jammy. This issue describes the required changes to support this new version. 

https://github.com/wazuh/wazuh/issues/13162

**Comment by pavlos1982** [comment]
all agents hows the no result for vulnerability:
![Screenshot 2022-05-09 075938 3](https://user-images.githubusercontent.com/5363713/167338497-1223f65a-0b30-40c3-9caa-dde2427091d6.jpg)

**Reply by MiguelazoDS** [reply]
Could you share the logs from the last full scan? please configure `wazuh_modules.debug=2` in your `/var/ossec/etc/internal_options.conf` file. The partial scan doesn't trigger alerts for already reported vulnerabilities.

**Reply by pavlos1982** [reply]
Hi MiguelazoDS, please see attached for the log
[wazuh-ossec-log.csv](https://github.com/wazuh/wazuh/files/8679082/wazuh-ossec-log.csv)
s

**Reply by MiguelazoDS** [reply]
Hi pavlos, 

I don't see anything bad in your logs. What I can see is all the vulnerabilities found were fixed by having the proper KB installed. 

I did a test in my environment and I see the same as you

![1](https://user-images.githubusercontent.com/13010397/168152318-96d749d8-034c-4bca-ab0b-3ea1c6472a6e.png)

But then installing a well-known vulnerable package I'm able to see the vulnerabilities alert. 

![2](https://user-images.githubusercontent.com/13010397/168152394-caa308f2-9894-4691-b5c1-307fe354f01c.png)

Do you think there's a vulnerability that should be reported and it is not? 

The scan on your manager is working as expected?

**Reply by pavlos1982** [reply]
Hi @MiguelazoDS ,

I believe you are right, I further added some more host to Wazuh and vulnerabilities was detected, thank you for taking to respond and help out.

Regards,

Paul

**Reply by MiguelazoDS** [reply]
I'm glad to know we could find an answer to this behavior. Have a great day!

**Reply by subhrajy** [reply]
> Hi pavlos,
> 
> I don't see anything bad in your logs. What I can see is all the vulnerabilities found were fixed by having the proper KB installed.
> 
> I did a test in my environment and I see the same as you
> 
> ![1](https://user-images.githubusercontent.com/13010397/168152318-96d749d8-034c-4bca-ab0b-3ea1c6472a6e.png)
> 
> But then installing a well-known vulnerable package I'm able to see the vulnerabilities alert.
> 
> ![2](https://user-images.githubusercontent.com/13010397/168152394-caa308f2-9894-4691-b5c1-307fe354f01c.png)
> 
> Do you think there's a vulnerability that should be reported and it is not?
> 
> The scan on your manager is working as expected?

Hi, this happened with me. Wazuh only displays vulnerabilities associated with the third-party packages and does not show the vulnerabilities associated with the OS itself like windows 11. 

Can you help with this?

**Reply by MiguelazoDS** [reply]
Hi @subhrajy, 

If your Windows agent is fully patched it is expected behavior, but we can take a look at the logs to be sure everything is working. 

What Wazuh version are you using? 

What is the output of these grep commands? 

`grep "scan will be run" /var/ossec/logs/ossec.log`
`grep -E "ERR|WARN|CRIT" /var/ossec/logs/ossec.log`

**Reply by subhrajy** [reply]
Hi, thanks for replying. 

yes, the windows agent is fully patched with all the available updates. 
I am using wazuh 4.3.7

I uninstalled one security update and wazuh actually gave me events / alerts on the vulnerability monitoring. So confirmed it is working fine. 

I will post the outputs if you need it for further study, once I get back to the workstation.

**Reply by MiguelazoDS** [reply]
Hi @subhrajy, 

If you could check the module is working as expected there's no need to share the logs, it was only to check the well-functioning of vulnerability scan. In case you find something else please don't hesitate to ask. Regards!
